### DISCUSSION TITLE
Disable IdleTimeout or create system PING\PONG

### DISCUSSION THREAD

**Discussion by Koldunz** [uNetworking/uWebSockets.js#119] [Q&A] [answered]
I use v15.9.0

Tell me how to disable automatic connection disconnection( disable idleTimeout ) or how to send system ping \ pong from server to clients using OpCode 0x9 [PING] and 0xA [PONG]

If the opportunity to implement events ping\pong?
`uWS.App().ws('/*', {
    ping: (ws) => {
        console.log("PING")
    },
    drain: (ws) => {
        console.log("PONG")
    }`

Thanks!

**Comment by ghost** [comment, marked as answer]
Set it to 0

**Reply by seiyria** [reply]
So if I understand this correctly, setting `idleTimeout` to 0 will disable automatic disconnects?

**Comment by roytan883** [comment]
@alexhultman  how to send system ping \ pong from server to clients ? If the clients is mobile phone, need server sending ping and receiving pong to check clients status.  Or is there anyway to set `heartbeat timeout` ?

**Comment by ghost** [comment]
You can't currently. Clients send messages to the server to keep alive

**Comment by RussCoder** [comment]
@alexhultman Also I want to point out that currently (I'm using v15.11.0) a connection is closed even if a message was sent by the server to the client, while the [documentation](https://unetworking.github.io/uWebSockets.js/generated/interfaces/websocketbehavior.html#idletimeout) says that the `idleTimeout` is:

> Maximum amount of seconds that may pass without **sending** or getting a message.

I suppose either the behavior should be changed or the documentation must be corrected. 

And I would like to have the ping/pong feature too.

**Comment by ghost** [comment]
@RussCoder You mind creating a new bug for that?

**Comment by RussCoder** [comment]
@alexhultman 
Did  it: https://github.com/uNetworking/uWebSockets.js/issues/187

**Comment by ghost** [comment]
Here you go:

Put this on the client.

```
let wsTimeout = null;
function ping(){
    let msg = {
        type: 'PING'
    }
    const $pingmsg = SOCKET.send(JSON.stringify(msg));
}
const $keepAlive = setInterval(ping, 50000);
```

**Comment by ghost** [comment]
Note that this discussion is slightly outdated; we are going to support sending pings from the server in an easy way in future releases.
